Update description of Fragments to emphasize evolving data needs (#1193)
graphql/graphql-spec: spec/Section 2 -- Language.md

Fragments are the primary unit of composition in GraphQL.

Fragments allow for the definition of modular selection sets that make it easier

> to add and remove selections as needed.

Reviewer: Maybe something along the lines of (written whilst on Zoom, not final wording!):

```
Fragments allow client components and logic to describe their specific
data requirements locally. Fragments can then be composed together to
form a GraphQL operation to issue to the server.
```

Fragments are the primary unit of composition in GraphQL.

> Fragments allow for the definition of selection sets that are colocated with the 
> logic that requires those selections, making it easier to add and remove selections 
> as needed.

Reviewer 2: Tentative suggestion that keeps the concept of "colocation" out of the spec for the time being.

```suggestion
Fragments allow grouping selections that belong together, making it easier to scope 
data accesses, add and remove selections as needed.
```

> themselves to form a GraphQL operation to issue to the server.

Reviewer: I'm concerned that "component" here will be read too specifically (e.g. to mean a React component) rather than broadly (e.g. a "component" of client logic such as a function, class, or other...). Perhaps something along the lines of:

```suggestion
Each data-consuming component (function, class, UI element, and so on) of
a client application should declare its data needs in a dedicated fragment.
These fragments may then be composed, following the usage of the components
themselves, to form a GraphQL operation to issue to the server.
```

}
```

> The fields required by `friendProfile` can be extracted into a fragment and

Reviewer: ```suggestion
The fields required by the `friendProfile` logic can be extracted into a fragment and
```

}
```

The fields required by `friendProfile` can be extracted into a fragment and 
composed by a parent fragment or operation.

```graphql example
query withFragments {
  user(id: 4) {
    friends(first: 10) {

>       ...friendProfileFragment

Reviewer 2: I personally like `friendProfile`, `Fragment` is making me type more for no obvious reason as I know it's a fragment from the context.
